File: shared_resources/src/main/java/com/thewizrd/shared_resources/helpers/ToggleAction.java
package com.thewizrd.shared_resources.helpers;

public final class ToggleAction extends Action {
    private boolean enabled;

    private ToggleAction() {
        super();
    }

    public ToggleAction(Actions action, boolean enabled) {
        super(action);
        this.enabled = enabled;
        actionSuccessful = true;
    }

    public boolean isEnabled() {
        return enabled;
    }

    public void setEnabled(boolean enabled) {
        this.enabled = enabled;
    }
}

File: shared_resources/src/main/java/com/thewizrd/shared_resources/actions/ToggleAction.kt
package com.thewizrd.shared_resources.actions

class ToggleAction(action: Actions, var isEnabled: Boolean) : Action(action) {
    init {
        isActionSuccessful = true
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is ToggleAction) return false

        if (isEnabled != other.isEnabled) return false

        return true
    }

    override fun hashCode(): Int {
        return isEnabled.hashCode()
    }
}